tests/roots/test-ext-autosectionlabel-full-reference/index.rst

```diff
@@ -0,0 +1,48 @@
+Introduction of Sphinx
+======================
+
+Installation
+============
+
+For Windows users
+-----------------
+
+Windows
+^^^^^^^
+
+Command
+'''''''
+
+For UNIX users
+--------------
+
+Linux
+^^^^^
+
+Command
+'''''''
+
+FreeBSD
+^^^^^^^
+
+Command
+'''''''
+
+This one's got an apostrophe
+----------------------------
+
+
+References
+==========
+
+* :ref:`index:Introduction of Sphinx`
+* :ref:`index:Installation`
+* :ref:`index:Installation:For Windows users`
+* :ref:`index:Installation:For Windows users:Windows`
+* :ref:`index:Installation:For Windows users:Windows:Command`
+* :ref:`index:Installation:For UNIX users`
+* :ref:`index:Installation:For UNIX users:Linux`
+* :ref:`index:Installation:For UNIX users:Linux:Command`
+* :ref:`index:Installation:For UNIX users:FreeBSD`
+* :ref:`index:Installation:For UNIX users:FreeBSD:Command`
+* :ref:`index:Installation:This one's got an apostrophe`
```

Comment:
Can you check check with numbers in titles as well as with `:math:` blocks please? and with literal (double backquotes). Can we also use `:ref:` in a title? (if not, no need for the coverage).

Additionally, can you test when you specify a custom label before an inner title? as well as content in-between two titles (in other words, try to see if it's possible to construct a doctree with the following tree: "section -> non-section -> section") so that the parent of the second section is a non-section node.

Reply:
Would adding an image in between suffice? e.g.

Section
-------

Here is an image or I could use a figure?

.. image:: test.png

Sub Section
===========

See previous image...

Reply:
Unfortunately, the image and subsection would be sibling nodes and not parent->child nodes. Maybe it's not possible to actually have a parent node that is not a section's node but I need to check that (it's midnight here so I won't do it today).

Reply:
I have tested with numbers in sections

Reply:
I have tried to added all the other comments, and added texinfo and latex tests and the test includes another rst file, to check how references worked there.
How is it looking now?